dollarskip.c

```diff
@@ -2,11 +2,103 @@
 #include <stdlib.h>
 #include <string.h>
 #include <assert.h>
+#include <stdbool.h>
 
 #ifndef COMMAND_MAX_SIZE
 #define COMMAND_MAX_SIZE 8096
 #endif
 
+/*** wrapped shell builtins ***/
+int easter_egg() {
+	printf("You found the DollarSkip easter egg!\n");
+	return 0;
+}
+
+/*** handled and unhandled shell builtins checks ***/
+const char *unhandled_shell_builtins[] = {
+	"cd",
+	"alias",
+	"unalias",
+	"pushd",
+	"popd",
+	"dirs",
+	"read",
+	"set",
+	"readonly",
+	"unset",
+	"typeset",
+	"declare",
+	"disown",
+	"enable",
+	"export",
+	"exit", // doesn't do anything when run through DollarSkip
+	"fc",
+	"history",
+	"fg",
+	"getopts",
+	"hash", // we don't want it to report fake data dor the current shell (true for the shell in which it is run)
+	"jobs", // same reason as the one for 'hash'
+	"times", // same as above
+	"local"
+};
+#define UNHANDLED_SHELL_BUILTINS_SIZE (sizeof(unhandled_shell_builtins) / sizeof(unhandled_shell_builtins[0]))
+
+typedef int (*wrapperFn)();
+
+typedef struct wrapper {
+	wrapperFn func;
+	const char *name; // name HAS to be a string literal
+} Wrapper;
+
+Wrapper handled_shell_builtins[] = {
+	{easter_egg, "e_egg"} // DollarSkip easter eff
```

Comment:
```suggestion
	{easter_egg, "e_egg"} // DollarSkip easter egg
```

What does this do exactly?